fdbclient/AuditUtils.actor.cpp

```diff
@@ -109,31 +268,62 @@ ACTOR Future<UID> persistNewAuditState(Database cx,
 	ASSERT(!auditState.id.isValid());
 	state Transaction tr(cx);
 	state UID auditId;
-
-	loop {
-		try {
-			tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
-			tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-			wait(checkMoveKeysLock(&tr, lock, ddEnabled, true));
-			std::vector<AuditStorageState> auditStates = wait(getLatestAuditStatesImpl(&tr, auditState.getType(), 1));
-			uint64_t nextId = 1;
-			if (!auditStates.empty()) {
-				nextId = auditStates.front().id.first() + 1;
+	state AuditStorageState latestExistingAuditState;
+	TraceEvent(SevDebug, "AuditUtilPersistedNewAuditStateStart", auditId);
+	try {
+		loop {
+			try {
+				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+				tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+				wait(checkMoveKeysLock(&tr, lock, ddEnabled, true));
+				RangeResult res =
+				    wait(tr.getRange(auditKeyRange(auditState.getType()), 1, Snapshot::False, Reverse::True));
+				ASSERT(res.size() == 0 || res.size() == 1);
+				uint64_t nextId = 1;
+				if (res.size() == 1) {
```

Comment:
super nit: `if (!res.empty())`, logically they are the same, but for readability, it is more clear we want to say that if there is a result, we want to do something.

Reply:
Changed.